internal/clients/api_client.go

```diff
@@ -307,5 +346,87 @@ func newEsApiClient(d *schema.ResourceData, key string, version string, useEnvAs
 		es.Transport = newDebugTransport("elasticsearch", es.Transport)
 	}
 
-	return &ApiClient{es, nil, version}, diags
+	return es, diags
+}
+
+func buildKibanaClient(d *schema.ResourceData, baseConfig BaseConfig, useEnvAsDefault bool) (*kibana.Client, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	kibConn, ok := d.GetOk("kibana")
+	if !ok {
+		return nil, diags
+	}
+
+	config := kibana.Config{}
+
+	// Use ES details by default
+	config.Username = baseConfig.Username
+	config.Password = baseConfig.Password
+	config.Address = baseConfig.Addresses[0]
```

Comment:
Kibana and ES won't share an address. It only really makes sense to share the credentials. 

```suggestion
	// Use ES details by default
	config := kibana.Config{
		Username: baseConfig.Username,
		Password: baseConfig.Password,
	}
```